test/test_spiner_transform.cpp

```diff
@@ -0,0 +1,95 @@
+#include <array>
+#include <cmath>
+#include <cstdio>
+#include <cstdlib>
+#include <iostream> // debug
+
+#include <ports-of-call/portability.hpp>
+#include <singularity-eos/eos/eos_spiner_common.hpp>
+#include <singularity-eos/eos/eos_spiner_sie_transforms.hpp>
+
+#include <spiner/databox.hpp>
+#include <spiner/interpolation.hpp>
+
+#ifndef CATCH_CONFIG_FAST_COMPILE
+#define CATCH_CONFIG_FAST_COMPILE
+#include <catch2/catch_test_macros.hpp>
+#endif
+
+#include <test/eos_unit_test_helpers.hpp>
+
+#ifdef SPINER_USE_HDF5
+#include <singularity-eos/base/spiner_table_utils.hpp>
+#endif
+
+#ifdef SPINER_USE_HDF5
+#ifdef SINGULARITY_TEST_SESAME
+
+using namespace singularity;
+using spiner_common::DataBox;
+using spiner_common::Grid_t;
+using spiner_common::to_log;
+
+PORTABLE_INLINE_FUNCTION bool isVeryClose(Real a, Real b, Real eps = 1e-14) {
+  return fabs(b - a) / (fabs(a + b) + 1e-20) <= eps;
+}
+
+struct TestDataContainer {
+  Real lRhoOffset = 0.5;
+
+  using RG = Spiner::RegularGrid1D<Real>;
+  Grid_t grid{{RG(1.0, 10.0, 100)}};
+  DataBox sieCold;
+
+  TestDataContainer() {
+    int N = static_cast<int>(grid.nPoints());
+    sieCold.resize(N);
+    sieCold.setRange(0, grid);
+    for (int i = 0; i < grid.nPoints(); ++i) {
+      Real lRho = grid.x(i);
+      Real e_cold = 3.0 * lRho + 5.0;
+      sieCold(i) = e_cold;
+    }
+  }
+};
```

Comment:
I think this code will benefit from some `static constexpr` constants. I'd also define a way to compute the cold curve as a function so it's accessible in the test.

Something like this:
```suggestion
struct TestDataContainer {
  static constexpr Real lRhoOffset = 0.5;
  static constexpr Real lRho_low = 1.0;
  static constexpr Real lRho_high = 10.0;
  static constexpr size_t Npts = 100;
  
  using RG = Spiner::RegularGrid1D<Real>;
  Grid_t grid{{RG(lRho_low, lRho_high, Npts)}};
  DataBox sieCold{Npts};

  TestDataContainer() {
    sieCold.setRange(0, grid);
    for (int i = 0; i < Npts; ++i) {
      const Real lRho = grid.x(i);
      sieCold(i) = e_cold_fun(lRho);
    }
  }
  static constexpr Real e_cold_fun(Real lRho) {
    return 3.0 * lRho + 5.0;
  }
};
```